### DISCUSSION TITLE
how to access the front matter for the current slide

### DISCUSSION THREAD

**Discussion by gureckis** [slidevjs/slidev#1671] [Q&A] [answered]
I have read the docs on [vue context](https://sli.dev/custom/vue-context) but it is still a little unclear to me how i can access the front matter of a particular slide.  for instance in global-top.vue i might want to show/hide information based on a front matter config
---
showlogo: false
---

then in global-top.vue

```
<div v-if='currentslide.frontmatter.showlogo'></div>
```

Is this possible?  I only see the front matter for the first slide.  Thanks!!

**Comment by gureckis** [comment]
found it!  `$slidev.nav.slides[$slidev.nav.currentPage].meta.slide.frontmatter`

**Comment by kermanx** [comment, marked as answer]
`$frontmatter` in fact

**Reply by gureckis** [reply]
hmmm... $frontmatter currently evaluates as an null object for me

e.g., here's my global-top.vue with the default project/theme (which has a lot of front matter for the first and subsequent slides)
```js
<script setup>
import { onMounted} from 'vue'
import { useSlideContext } from '@slidev/client'
const { $frontmatter } = useSlideContext()

onMounted(() => {
  console.log($frontmatter)
})

</script>
```

output is `{}` (this is v0-.49.11)

**Reply by kermanx** [reply]
Sorry for the ambiguities in my answer. Global layers like global-top.vue, like its name, is global, which doesn't specify to one slide. About using slide-top.vue which applies to each slide instead? The usage of Slide layers is the same as Global layers, except for the file name.

**Reply by gureckis** [reply]
I finally got around to checking into this.  Even in slide-bottom.vue $frontmatter is empty for me in 0.49.13

```vue
<script setup>
import { ref, onMounted } from 'vue'
import { useSlideContext } from '@slidev/client'
const { $slidev, $frontmatter, $page } = useSlideContext()

console.log($frontmatter)
onMounted(() => {console.log($frontmatter)})
</script>

**Reply by gureckis** [reply]
possibly this line is a typo (fontmatter instead of frontmatter)... there are several references to 'fontmatter' in the code that might be typos. 
 https://github.com/slidevjs/slidev/blob/e2640eef529bf096d8ad2fdc26c8561866aabf5e/packages/client/constants.ts#L14

**Reply by kermanx** [reply]
My bad. This is a bug (not only the typo). Sorry for the misleading message😭
